hack/cpaas/patch-cluster-with-rhpm-0.1.sh

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env bash
+set -eu -o pipefail
+
+PROJECT_ROOT="$(git rev-parse --show-toplevel)"
+declare -r PROJECT_ROOT
+declare -r powermon_package_name="power-monitoring-operator-bundle-container"
+declare -r OCP_VERSION=${OCP_VERSION:-'v4.13'}
+
+source "$PROJECT_ROOT/hack/utils.bash"
+
+function check_prereq {
+  header "checking pre requisites..."
+  if ! command -v jq &> /dev/null
+  then
+      err "jq could not be found. Please install jq first."
+      exit
+  fi
+  if ! command -v oc &> /dev/null
+  then
+      err "oc could not be found. Please install oc first."
+      exit
+  fi
+  if ! command -v podman &> /dev/null
+  then
+      err "podman could not be found. Please install podman first."
+      exit
+  fi
+  ok "all prereqs ok.."
+}
```

Comment:
This can be replaced by having an `ensure_all_tools` func like this:
```
declare -r TMP_DIR="$PROJECT_ROOT/tmp"
declare -r BIN_DIR="$TMP_DIR/bin"

ensure_all_tools() {
	header "Ensuring all tools are installed"
	"$PROJECT_ROOT/hack/tools.sh" all
}
...
...

main(){
     export PATH="$BIN_DIR:$PATH"
     ensure_all_tools
...
...
}
```

Reply:
it is ok here as wel.

Reply:
well, having `ensure_all_tools` will always make sure that necessary tools are available before the script executes.

Reply:
same is done by this function. i am not that keen on changing this.